src/test/java/org/broadinstitute/hellbender/utils/nio/NioTests.java

```diff
@@ -0,0 +1,118 @@
+    }
+
+
+    /**
+     * When we give no explicit credentials, then the system will still work if either
+     * - the GOOGLE_APPLICATION_CREDENTIALS environment variable is set
+     * - the user ran "gcloud auth login" first
+     * - the code is running on a Google Cloud Compute machine.
+     * (see http://gcloud-python.readthedocs.org/en/latest/gcloud-auth.html)
+     */
+    @Test(groups = {"cloud"})
+    public void openPrivateFileUsingDefaultCredentials() throws IOException {
+        // this file, potentially unlike the others in the set, is not marked as "Public link".
+        final String privateFile = getGCPTestInputPath() + "org/broadinstitute/hellbender/utils/private_file.txt";
+
+        Path path = Paths.get(URI.create((privateFile)));
+        int firstByte = Files.newInputStream(path).read();
+    }
+
+    /**
+     * Opening the private file even when the user is not logged in on gcloud should work
+     * when we provide explicit credentials.
+     */
+    @Test(groups = {"cloud"})
+    public void openPrivateFileWithExplicitCredentials() throws IOException {
+        // this file, potentially unlike the others in the set, is not marked as "Public link".
+        final String privateFile = getGCPTestInputPath() + "org/broadinstitute/hellbender/utils/private_file.txt";
+        final String[] split = privateFile.split("/");
+        final String BUCKET = split[2];
+        final String pathWithoutBucket = String.join("/", Arrays.copyOfRange(split, 3, split.length));
+
+        FileSystem fs = getAuthenticatedGcs(BUCKET);
+        Path path = fs.getPath(pathWithoutBucket);
+        int firstByte = Files.newInputStream(path).read();
+    }
+
+    /**
+     * Using explicit credentials only works on that access, they are not kept.
+     * This test will fail if default credentials are available.
+     */
+    @Test(enabled = false, groups = {"cloud"}, expectedExceptions = {StorageException.class})
+    public void explicitCredentialsAreNotKept() throws IOException {
+        // this file, potentially unlike the others in the set, is not marked as "Public link".
+        final String privateFile = getGCPTestInputPath() + "org/broadinstitute/hellbender/utils/private_file.txt";
+        final String privateFile2 = getGCPTestInputPath() + "org/broadinstitute/hellbender/utils/private_file_2.txt";
+        final String[] split = privateFile.split("/");
+        final String BUCKET = split[2];
+        final String pathWithoutBucket = String.join("/", Arrays.copyOfRange(split, 3, split.length));
+
+        FileSystem fs = getAuthenticatedGcs(BUCKET);
+        Path path = fs.getPath(pathWithoutBucket);
+        int firstByte = Files.newInputStream(path).read();
+
+        // now let's open another private file. It shouldn't work.
+        path = Paths.get(URI.create((privateFile2)));
+        firstByte = Files.newInputStream(path).read();
+    }
+
+    private FileSystem getAuthenticatedGcs(String bucket) throws IOException {
```

Comment:
So I went back and split this in two. Left the test-specific bits here, and moved the rest to BucketUtils. That takes a byte[] with the credentials, so we can pass them around.

When running on Dataproc, though, default credentials work and so we don't need to do this we can just open the file with CloudStorageFileSystem.forBucket(bucket).